Theory guide cfd dem (#461)
lethe-cfd/lethe: doc/source/theory/unres_cfd-dem/unres_cfd-dem.rst

.. image:: images/schematic_unresolve_cfd-dem.jpg
    :alt: Schematic represantion of micro-meso scale approach in unresolved CFD-DEM
    :align: center
    :name: geometry

In this guide, we summarize the theory behind Unresolved CFD-DEM. For further details, we refer the reader to the articles by Bérard, Patience & Blais (2019) `[1] <https://doi.org/10.1002/cjce.23686>`_, and Zhou et al. (2010) `[2] <https://doi.org/10.1017/S002211201000306X>`_.

Particles
----------

Applying Newton's second law on the particle :math:`i` surrounded by fluid a :math:`f`, we find:

.. math::
    m_i \frac{d \mathbf{v}_i}{dt} = \sum_{j}\mathbf{f}_{c,ij} + \sum_{j}\mathbf{f}_{nc,ij} + \mathbf{f}_{pf,i} + \mathbf{f}_{g,i} \\
    I_i \frac{d\mathbf{\omega}_i}{dt} = \sum_{j}\left ( \mathbf{M}_{c,ij} + \mathbf{M}_{r,ij} \right ) + \sum_{w}\left ( \mathbf{M}_{c,iw} + \mathbf{M}_{r,iw} \right )

where:

* :math:`m_i` is the mass of the particle;
* :math:`v_i` is the velocity vector;
* :math:`\mathbf{f}_{c,ij}` are the contact forces between particles :math:`i` and :math:`j` (detailed in the DEM section of this guide);

> * :math:`\mathbf{f}_{nc,ij}` are the non-contact forces between particles :math:`i` and :math:`j`, such as lubrication `[3] <https://doi.org/10.1002/aic.690400418>`_.

Reviewer: I would also add a note saying that no such forces are implemented yet in Lethe.

> * :math:`\mathbf{f}_{nc,ij}` are the non-contact forces between particles :math:`i` and :math:`j`, such as lubrication `[3] <https://doi.org/10.1002/aic.690400418>`_.

Author: Added it. Is it good as it is now?

where:

* :math:`\mathbf{f}_{\nabla p,i}` is the force due to the pressure gradient;
* :math:`\mathbf{f}_{\nabla \cdot \tau,i}` is the force due to the shear stress;
* :math:`\mathbf{f}_{d,i}` is the drag force;
* :math:`\mathbf{f}_{Ar,i}` is the buoyant (Archimedes) force;
* :math:`\mathbf{f}_{g,i}` is the force due to gravity;
* :math:`\mathbf{f}''_{i}` are the remaining forces, including virtual mass, Basset, Lift, and Magnus (currently not implemented in Lethe).

.. note::
    Since pressure in Lethe does not account for the hydrostatic pressure, i.e., the gravity term is not taken into account in the Navier-Stokes equations (see :doc:`../fluid_dynamics/navier-stokes`), we explicitly insert :math:`\mathbf{f}_{Ar,i}` in :math:`\mathbf{f}_{pf,i}`.

In unresolved CFD-DEM drag is calculated using correlations (frequently called drag models). The drag models implemented in Lethe are described in the `unresolved CFD-DEM parameters guide <https://lethe-cfd.github.io/lethe/parameters/unresolved_cfd-dem/cfd_dem.html>`_.

Volume Average Navier-Stokes
-----------------------------

> Since we represent the fluid at a meso-scale, the quantities calculated for the subdomains are averages among its volume. Additionally, as the volume of fluid is a fraction of the subdomain, the porosity (or void fraction) is taken into account. To do this, we apply the Volume Average Navier-Stokes (VANS) equations to represent the fluid phase. Mainly, the VANS equations are presented in two different formulations, so called Model A (or Set II) and Model B (or Set I).

Reviewer: ```suggestion
Since we represent the fluid at a meso-scale, the quantities calculated for the subdomains are averages among its volume. Additionally, as the volume of fluid is a fraction of the subdomain, the porosity (or void fraction) is taken into account. To do this, we apply the Volume Average Navier-Stokes (VANS) equations to represent the fluid phase. Mainly, the VANS equations are presented in two different formulations, so called Model A (or Set II) and Model B (or Set I).
```
What is meant by subdomain here? You mean cells or you mean phases? This is not clear.

Author: I changed 'subdomains' for 'cells'. I think it is more clear now. What do you think?

> For both models, considering incompressible fluid, the continuity is:

Reviewer: ```suggestion
Considering an incompressible flow, the continuity equation for both models is:
```
Be careful, there are no such things as incompressible fluids. A flow of water can be compressible or incompressible, same for gas. Being incompressible is a property of a flow.

Model B:

.. math:: 
    \rho_f \left ( \frac{\partial \left ( \varepsilon_f \mathbf{u} \right )}{\partial t} + \nabla \cdot \left ( \varepsilon_f \mathbf{u} \otimes \mathbf{u} \right ) \right ) = -\nabla p + \nabla \cdot \tau + \mathbf{F}_{pf}^B

where:

* :math:`\rho_f` is the density of the fluid;
* :math:`p` is the pressure;
* :math:`\tau` is the shear stress;
* :math:`\mathbf{F}_{pf}^A` and :math:`\mathbf{F}_{pf}^B` are the source terms representing the forces applied back in the fluid due to the interaction with particles for Models A and B, respectively.

For Model A, since the pressure term corresponds to a 'fluid fraction of the pressure', we can write the interaction term as:

.. math:: 
    \mathbf{F}_{pf}^A = -\frac{1}{V_{\Omega}}\sum_{i}^{n_p}\left ( \mathbf{f}_{pf, i} - \mathbf{f}_{\nabla p, i} - \mathbf{f}_{\nabla \cdot \tau, i} \right )

while for Model B, since the pressure is totally in the fluid, we write:

.. math:: 
    \mathbf{F}_{pf}^B = -\frac{1}{V_{\Omega}}\sum_{i}^{n_p}\left ( \mathbf{f}_{pf, i} \right )

> where :math:`n_p` is the number of particles inside the subdomain :math:`\Omega` with volume :math:`V_{\Omega}`.

Reviewer: ```suggestion
where :math:`n_p` is the number of particles inside the subdomain :math:`\Omega` with volume :math:`V_{\Omega}`.
```
I guess here subdomain means cell? If that's what you mean, please define it properly before (like significantly above) and explain that in what follows a subdomain is generally a cell.

Lethe is capable of simulating unresolved CFD-DEM cases with both Models A and B (see the :doc:`../../parameters/unresolved_cfd-dem/cfd_dem` page of this guide).

Void fraction
--------------
Determining the void fraction is an important step in unresolved CFD-DEM, as can be noted by the VANS equations and the drag models `[4] <http://dx.doi.org/10.1016/j.ces.2013.05.036>`_. In Lethe we apply the Particle Centroid Method (PCM) `[5] <https://doi.org/10.1002/aic.14421>`_, which is simple and the most popular method. It consists of tracking the position of the centroid of each particle and applying the total volume of the particle on the calculation of the void fraction of the cell. This means that in either of the following situations the void fraction of the colored cell is the same:

.. image:: images/void_frac1.png
.. image:: images/void_frac2.png

PCM can be written as:

.. math:: 
    \varepsilon_f = 1 - \frac{\sum_{i}^{n_p} V_{p,i}}{\Omega}

where :math:`n_p` is the number of particles with centroid inside the subdomain :math:`V_{\Omega}`.

> In finite Element, the void fraction must be projected to the mesh nodes so that one can assemble the system of equations. This is done by :math:`\mathcal{L}^2` projection `[6] <https://link.springer.com/book/10.1007/978-3-642-33287-6>`_:

Reviewer: ```suggestion
In the finite element method, the void fraction must be projected to the mesh nodes so that one can assemble the system of equations. This is done by :math:`\mathcal{L}^2` projection `[6] <https://link.springer.com/book/10.1007/978-3-642-33287-6>`_:
```
